omnigibson/systems/macro_particle_system.py

```diff
@@ -343,67 +327,94 @@ def particle_idn2name(cls, idn):
-
-    @classmethod
-    def initialize(cls):
+    def __init__(
+        self, name, create_particle_template, min_scale=None, max_scale=None, scale_relative_to_parent=False, **kwargs
+    ):
+        """
+        Args:
+            name (str): Name of the visual particles, in snake case.
+            create_particle_template (function): Method for generating the visual particle template that will be duplicated
+                when generating groups of particles.
+                Expected signature:
+
+                create_particle_template(prim_path: str, name: str) --> EntityPrim
+
+                where @prim_path and @name are the parameters to assign to the generated EntityPrim.
+                NOTE: The loaded particle template is expected to be a non-articulated, single-link object with a single
+                    visual mesh attached to its root link, since this will be the actual visual mesh used
+            min_scale (None or 3-array): If specified, sets the minumum bound for the visual particles' relative scale.
+                Else, defaults to 1
+            max_scale (None or 3-array): If specified, sets the maximum bound for the visual particles' relative scale.
+                Else, defaults to 1
+            scale_relative_to_parent (bool): If True, will scale generated particles relative to the corresponding
+                group's object
+            **kwargs (any): keyword-mapped parameters to override / set in the child class, where the keys represent
+                the class attribute to modify and the values represent the functions / value to set
+                (Note: These values should have either @classproperty or @classmethod decorators!)
+        """
+        self._scale_relative_to_parent = scale_relative_to_parent
+        self._create_particle_template_fcn = create_particle_template
+
+        # Maps particle name to dict of {obj, link, face_id}
+        # NOTE: link will only exist for particles on rigid bodies
+        # NOTE: face_id will only exist for particles on cloths
+        self._particles_info = None
+
+        # Pre-cached information about visual particles so that we have efficient runtime computations
+        # Maps particle name to local pose matrix for computing global poses for the particle
+        self._particles_local_mat = None
+
+        # Maps group name to array of face_ids where particles are located if the group object is a cloth type
+        # Maps group name to np.array of face IDs (int) that particles are attached to
+        self._cloth_face_ids = None
+
+        # Default behavior for this class -- whether to clip generated particles halfway into objects when sampling
+        # their locations on the surface of the given object
+        self._CLIP_INTO_OBJECTS = False
+
+        # Default parameters for sampling particle locations
+        # See omnigibson/utils/sampling_utils.py for how they are used.
+        self._SAMPLING_AXIS_PROBABILITIES = (0.25, 0.25, 0.5)
```

Comment:
now that this is a proper class, we casn probably just make these normal kwargs in the __init__ call, what do you think?

Reply:
TODO: do this